src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/DeleteCompleteFilesTask.java

```diff
@@ -80,58 +85,72 @@ private void processCompleteFiles(CloudBlobContainer container) {
 
         log.info(
             "Finished deleting complete files in container {}, deleted {} files, failed to delete {} files",
-            container.getName(),
+            container.getBlobContainerName(),
             successCount,
             failureCount
         );
     }
 
-    private boolean tryProcessCompleteEnvelope(CloudBlobContainer container, Envelope envelope) {
-        boolean deleted = false;
+    private boolean tryProcessCompleteEnvelope(BlobContainerClient container, Envelope envelope) {
 
-        String loggingContext = "File name: " + envelope.getZipFileName() + ", Container: " + container.getName();
+        String loggingContext = "File name: " + envelope.getZipFileName()
+            + ", Container: " + container.getBlobContainerName();
 
         try {
             log.info("Deleting file. {}", loggingContext);
 
-            CloudBlockBlob cloudBlockBlob = container.getBlockBlobReference(envelope.getZipFileName());
+            BlobClient blobClient = container.getBlobClient(envelope.getZipFileName());
 
-            if (cloudBlockBlob.exists()) {
-                Optional<String> leaseId = blobManager.acquireLease(
-                    cloudBlockBlob,
-                    container.getName(),
-                    envelope.getZipFileName()
+            if (Boolean.TRUE.equals(blobClient.exists())) {
+
+                leaseAcquirer.ifAcquiredOrElse(
+                    blobClient,
+                    leaseId -> delete(blobClient, leaseId),
+                    DeleteCompleteFilesTask::throwBlobDeleteException,
```

Comment:
looks like failure case is caught and logged in `delete` function. perhaps we just can get rid of `if exists` statement and just use this line for logging information that blob has already been deleted or smth

Reply:
you mean instead of checking existence we can directly try to delete ?

Reply:
Yea i am hoping we can avoid making extra call and just try delete straight away

Reply:
but in that case we depend on azure exceptions which I do not want now. Lets see how this goes and we can refactor later.

Reply:
sounds good to me. still - does it need exception throw? it wasn't throwing before

Reply:
in this case it should throw,  we mark the file deleted if there is no error.